src/MIDebugEngine/MIDebugCommandDispatcher.cs

```diff
@@ -27,7 +27,46 @@ public static Task<string> ExecuteCommand(string command)
 
                 lastProcess = s_processes[s_processes.Count - 1];
             }
-            return ExecuteCommand(command, lastProcess);
+
+            if (lastProcess == null)
+            {
+                throw new InvalidOperationException(MICoreResources.Error_NoMIDebuggerProcess);
+            }
+
+            return lastProcess;
+        }
+
+        public static MICore.ProcessState GetProcessState()
+        {
+            return GetLastProcess().ProcessState;
+        }
+
+        public static async Task<Results> SetSelectedThread(int threadId)
+        {
+            GetLastProcess().MICommandFactory.DefineCurrentThread(threadId);
+            return await Task.FromResult<Results>(new Results(ResultClass.done));
+        }
+
+        public static async Task<Results> ExecuteMICommandWithResultsObject(string command)
+        {
+            if (string.IsNullOrWhiteSpace(command))
+                throw new ArgumentNullException(nameof(command));
+
+            command = command.Trim();
+
+            if (command[0] == '-')
```

Comment:
Should this throw `ArgumentOutOfRange` instead?

Reply:
This method enables a user to obtain the Results object without stringification that is done inside the method ExecuteMiCommand.
Please see:
https://github.com/microsoft/MIEngine/blob/782c97ded5747639a947e094f8e9d7d06c153cf4/src/MIDebugEngine/MIDebugCommandDispatcher.cs#L84-L87

https://github.com/microsoft/MIEngine/blob/782c97ded5747639a947e094f8e9d7d06c153cf4/src/MICore/MIResults.cs#L582

Reply:
Sorry, what I mean is, instead of:

```
   if (command[0] != '-') return null;
```

I think this should be:
```
if (command[0] != '-') throw new ArgumentOutOfRangeException(nameof(command));
```